### DISCUSSION TITLE
Z2M 2.0.0, legacy triggers, _action entity's and hue dimmer blueprints

### DISCUSSION THREAD

**Discussion by popy2k14** [Koenkk/zigbee2mqtt#24997] [General]
Hi guys.

I have read about upcoming Z2M 2.0.0 which removes legacy features here: https://github.com/Koenkk/zigbee2mqtt/discussions/24198

I have already all legacy features set to false, except "homeassistant_legacy_triggers".

I using them (_action entities) extensively in my setup with 25x hue dimmer switches and the blueprint: https://epmatt.github.io/awesome-ha-blueprints/docs/blueprints/controllers/philips_929002398602/

Love that blueprint, because you can easily setup actions for an switch.

Sadly, the blueprint is using sensor.*_action as an input,
which will be removed in Z2M 2.0.0 and must be replaced with Matt device triggers, see here:
https://www.zigbee2mqtt.io/guide/usage/integrations/home_assistant.html#via-mqtt-device-trigger-recommended

So my questions are:
 * Am I right, my setup is affected by Z2M 2.0.0 is removing legacy features?
 * What blueprints/add-ons/custom-addons/automations/... do you use as a replacement to be Z2M 2.0.0 compatible (and not using _action)?

Thx

**Comment by Koenkk** [comment]
> Am I right, my setup is affected by Z2M 2.0.0 is removing legacy features?

Yes, definitely, all `_action` sensors are removed.

**Reply by popy2k14** [reply]
thx a lot.
Any recommendation what to use for easy configuration of hue dimmer switches (or any other zigbee remote) without creating 8+ Automations?

**Reply by popy2k14** [reply]
I am using this blueprints, and here is the same question in this thread:
https://community.home-assistant.io/t/awesome-ha-blueprints-a-curated-list-of-blueprints-easily-create-controller-based-automations-remotes-switches-for-controlling-lights-media-players-and-more/256687/106?u=popy

Just for reference, if someone has the same issue.

**Reply by MaJerle** [reply]
@popy2k14 I've posted the updated Blueprint that uses directly the MQTT topic subscription, and extracts action from the json payload.
https://community.home-assistant.io/t/philips-hue-dimmer-switch-v2-zigbee2mqtt/500610/39?u=tilz0r

On a long run we may need to move to `event.xxx` type of triggers

**Reply by mundschenk-at** [reply]
`event` entity support is currently experimental (i.e. some fields may change in the future), but you can already enable it on 1.42.0.

**Reply by popy2k14** [reply]
@MaJerle thx for your blueprint, already testing it. When i got your blueprint right it will trigger directly on action json payload.
Will this action payload also be removed in 2.0.0?

Sorry for asking, but i dont want to redo the whole thing again when 2.0.0 is released.

**Reply by popy2k14** [reply]
ok, tested it and set `homeassistant_legacy_triggers: false` - your blueprint is working and the old one not.
Will migrate over my 24x hue dimmer switches to your blueprint.

P.S.: Also my feelings are like yours (stated in your HA thread), the reaction is slightly faster.

thx a lot.

**Reply by MaJerle** [reply]
I've disabled all legacy features in my Z2M config and it all remains working for the time being. 
Long run, I'd like to move to the preferred Z2M solution, which seems to be `event.xxx`, but it seems there is a note it may change. 

Up until we have the stable release, I won't touch my current config.

**Reply by popy2k14** [reply]
Here are two other blueprints for the ikea e1766 and Aqara/Xiaomi Wireless mini switch WXKG11LM:

e1766: https://community.home-assistant.io/t/zha-deconz-zigbee2mqtt-ikea-e1766-tradfri-open-close-remote-universal-blueprint-all-actions-control-lights-media-players-and-more-with-hooks/351383/2?u=popy

Aqara/Xiaomi Wireless mini switch WXKG11LM: https://community.home-assistant.io/t/aqara-xiaomi-wireless-mini-switch-wxkg11lm/785944/3?u=popy

@MaJerle i have used your as a starting point, hope this is ok for you :-)
